### DISCUSSION TITLE
Begin Data Queries

### DISCUSSION THREAD

**Discussion by alexluong** [beginner-corp/begin-community#88] [Q&A] [answered]
Hi there! I'm new to Begin Data and DynamoDB as a whole and would like to learn more about queries. I'm familiar with MongoDB and general SQL *where* I can do queries like `where x equals y` or `where x is larger than y`. I have heard of DBs like DynamoDB where it's possible to model the data for patterns like these with partition keys and sort keys. I'm just wondering if it's possible to do the same with Begin Data and if there's any existing guide on them.

I've heard of Begin for a long time, recently checked it out again, and found it very very interesting. I'm hoping to build some side projects with it, and DBs are the only concern I have. Would love any guidance and pointers to resources I can use to learn more! Thanks a lot.

PS: as a primarily front end dev, every time I want to get hyped for serverless I usually find a podcast with Brian as a guest and just take in all this knowledge. Thanks and keep doing what yall are doing!

**Comment by brianleroux** [comment, marked as answer]
Hey! Sorry for the delayed response; `@begin/data` is really just a high level wrapper for working with a single DynamoDB table. You query things by `key`. This means you need to model things up front so you can query them later which is a bit different from other systems like RDBMS. If you have more specific use case I can help you model it to query!

**Reply by alexluong** [reply]
Thanks for the response!

This is quite fundamental, but if you can help with this, I think I can figure out other aspects.

How would you model a simple comparison query? For example:

```ts
interface Product {
  key: string;
  price: number;
}
```

How can I query products whose price is smaller than x (say, 10)? Would love if you can give some code example with `@begin/data` specifically!

**Reply by brimbey** [reply]
Hi @alexluong - did you ever make headway with this? I am having similar issues.

**Reply by alexluong** [reply]
Unfortunately no, I ended up not exploring Begin and DynamoDB much further.

**Reply by brianleroux** [reply]
lmk if this is helpful https://github.com/brianleroux/begin-data-crudl-example
